Remove not supported framework

eddynaka opened the pull request:
Fix Issue #1767

## Changes
- removing netcoreapp1.0 / 1.1 from targetframework
- removing netstandard1.3/1.6 and replace for netstandard2.0

### Checklist
- [X] I ran Unit Tests locally.
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed

The PR will trigger build, unit tests, and functional tests automatically. Please follow [these](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### Notes for authors:
- FxCop and other analyzers will fail the build. To see these errors yourself, compile localy using the Release configuration.

### Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

TimothyMothra commented:
I would ask that you revert your changes to Perf and lets address those in a follow-up PR.
That effort should be different from changing the target frameworks.

eddynaka commented:
In the build pipeline yml, we have this:
https://github.com/microsoft/ApplicationInsights-dotnet/blob/3ac8c0b14466247263271a70ee5cbf3f54437145/NETCORE/.vsts/linux-build.yml#L26-L32

I couldn't find that project. Was that renamed or removed?

cijothomas commented:
> In the build pipeline yml, we have this:
> https://github.com/microsoft/ApplicationInsights-dotnet/blob/3ac8c0b14466247263271a70ee5cbf3f54437145/NETCORE/.vsts/linux-build.yml#L26-L32
> 
> I couldn't find that project. Was that renamed or removed?

Yes, renamed - https://github.com/microsoft/ApplicationInsights-dotnet/pull/1824

cijothomas commented on BASE/Test/ServerTelemetryChannel.Test/TelemetryChannel.Nuget.Tests/TelemetryChannel.Nuget.Tests.csproj:
```diff
@@ -20,7 +20,7 @@
 
   <PropertyGroup Condition="$(OS) != 'Windows_NT'">
```
this whole section need to be removed. This test is only required for net45. To avoid running them in linux, we can rename project to .Win.Tests.csproj, and use "*Win.Tests.proj" as a filter in Linux build yml file.
This would apply to any other test project which are only supposed to targe full framework like .NET45.
(This can be done separately from this PR)

eddynaka replied:
I removed that section. and later, as you suggested, we can check a new pipeline in windows.
Thanks for reviewing

eddynaka commented:
> > In the build pipeline yml, we have this:
> > https://github.com/microsoft/ApplicationInsights-dotnet/blob/3ac8c0b14466247263271a70ee5cbf3f54437145/NETCORE/.vsts/linux-build.yml#L26-L32
> > 
> > I couldn't find that project. Was that renamed or removed?
> 
> Yes, renamed - #1824

Thanks. Just updated the pipeline!

TimothyMothra commented:
> > > In the build pipeline yml, we have this:
> > > https://github.com/microsoft/ApplicationInsights-dotnet/blob/3ac8c0b14466247263271a70ee5cbf3f54437145/NETCORE/.vsts/linux-build.yml#L26-L32
> > > 
> > > I couldn't find that project. Was that renamed or removed?
> > 
> > 
> > Yes, renamed - #1824
> 
> Thanks. Just updated the pipeline!

Eddy, this same file:
https://github.com/microsoft/ApplicationInsights-dotnet/blob/3ac8c0b14466247263271a70ee5cbf3f54437145/NETCORE/.vsts/linux-build.yml#L62

This is another test that we renamed. Can you please fix this here too and then confirm in the build log that it runs tests. If you look at the log now you'll see that it's empty

TimothyMothra commented on BASE/Test/ServerTelemetryChannel.Test/TelemetryChannel.Nuget.Tests/TelemetryChannel.Nuget.Tests.csproj:
```diff
@@ -18,11 +18,6 @@
-    <TargetFrameworks>netstandard1.3</TargetFrameworks>
-  </PropertyGroup>
-  
   <ItemGroup Condition="$(OS) == 'Windows_NT'">
```
I agree with this change, but it's causing the build to fail.
For now, I think you could revert it and just add a comment inside the PropertyGroup:
`<!-- This can be removed. Requires update to .yml -->`

You can create a github issue to investigate a fix later.

eddynaka replied:
ok, done: https://github.com/microsoft/ApplicationInsights-dotnet/issues/1847

eddynaka commented:
> Eddy, this same file:
> https://github.com/microsoft/ApplicationInsights-dotnet/blob/3ac8c0b14466247263271a70ee5cbf3f54437145/NETCORE/.vsts/linux-build.yml#L62
> 
> This is another test that we renamed. Can you please fix this here too and then confirm in the build log that it runs tests. If you look at the log now you'll see that it's empty

Sure, I can change that as well. For the part of 20, we was running .netcoreapp2.x, right now, our two integrationtests (normal and web one) targets netcoreapp3.1. Should i:
- target 2.1 as well
- remove that part

What about integrationTests.Web, should I add here as well?

thanks for reviewing
